apps/bors_worker/test/batcher/message_test.exs

```diff
@@ -50,16 +50,51 @@ defmodule BorsNG.Worker.BatcherMessageTest do
   end
 
   test "generate commit message" do
-    expected_message = "Merge #1 #2\n\n1: Alpha r=r\na\n\n2: Beta r=s\nb\n"
+    expected_message = "Merge #1 #2\n\n1: Alpha r=r\n\na\n\n2: Beta r=s\n\nb\n"
     patches = [
       %{patch: %{pr_xref: 1, title: "Alpha", body: "a"}, reviewer: "r"},
       %{patch: %{pr_xref: 2, title: "Beta", body: "b"}, reviewer: "s"}]
     patches2 = [
       %{patch: %{pr_xref: 2, title: "Beta", body: "b"}, reviewer: "s"},
       %{patch: %{pr_xref: 1, title: "Alpha", body: "a"}, reviewer: "r"}]
-    actual_message = Message.generate_commit_message(patches)
+    actual_message = Message.generate_commit_message(patches, nil)
     assert expected_message == actual_message
-    actual_message2 = Message.generate_commit_message(patches2)
+    actual_message2 = Message.generate_commit_message(patches2, nil)
     assert expected_message == actual_message2
   end
+
+  test "cut body" do
+    assert "a" == Message.cut_body("abc", "b")
+  end
+
+  test "cut commit message bodies" do
+    expected_message = """
+    Merge #1
+
+    1: Synchronize background and foreground processing r=bill
+
+    Fixes that annoying bug.
+    """
+    title = "Synchronize background and foreground processing"
+    body = """
+    Fixes that annoying bug.
+    
+    <!-- boilerplate follows -->
+    
+    Thank you for contributing to my awesome OSS project!
+    To make sure your PR is accepted ASAP, make sure all of this
+    stuff is done:
+    
+    - [ ] Run the linter
+    - [ ] Run any new or changed tests
+    - [ ] This PR fixes #___ (fill in if it exists)
+    - [ ] Make sure your commit messages make sense
+    """
+    patches = [
+      %{patch: %{pr_xref: 1, title: title, body: body}, reviewer: "bill"} ]
+    actual_message = Message.generate_commit_message(
+      patches,
+      "\n\n<!-- boilerplate follows -->")
+    assert expected_message == actual_message
+  end
 end
```

Comment:
For completeness, I would add a test for more than one match and another with no matches at all.